x/storage/keeper/keeper.go

```diff
@@ -66,6 +68,7 @@ func NewKeeper(
 		crossChainKeeper:   crossChainKeeper,
 		virtualGroupKeeper: virtualGroupKeeper,
 		authority:          authority,
+		chainID:            chainID,
```

Comment:
We can get the chainid directly from the context of the transaction.

```
	ctx := sdk.UnwrapSDKContext(goCtx)

	ctx.ChainID()
```

Reply:
fixed